### PR TITLE
correcting projects doc

### LINKED ISSUE #8077: [BUG] mlflow run using Docker container not creating /mlflow/projects/code folder and moving proper files to it
### Issues Policy acknowledgement

- [X] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Willingness to contribute

Yes. I would be willing to contribute a fix for this bug with guidance from the MLflow community.

### MLflow version

Mlflow version: 2.2.2 (but this bug was found in previous 2.x versions)
(mlflow server not in use)

### System information

- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
NAME="Ubuntu"
VERSION="18.04.6 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.6 LTS"
VERSION_ID="18.04"

- **Python version**:
Local: 3.10.9
Container: 3.7.

- **yarn version, if running the dev UI**:
(not having trouble with the UI or using it in this case)

### Describe the problem

Hi, I've been using mlflow 1.30.0 for ML pipeline orchestration without any issues. But when I tried using more recent versions of it (2.x) I faced troubles when running pipeline components using Docker as environment manager.

The problem: when I start a run (through "mlflow run [component path]", without any further options) mlflow just can't find my code in the directory it should be placed  inside the container (/mlflow/projects/code). Inspecting the created container I could confirm that the no code was moved to the container neither the folder was created.

When I switch back to version 1.30.0 (pip install mlflow==1.30.0) it immediately works.

### Tracking information

```shell
REPLACE_ME
```

### Code to reproduce issue

Since it is not a code related problem (it just can't even reach my code), I'll show my Docker file and my MLproject file, but it's needed to say that I tested it with different kinds of MLproject files and docker containers and encountered the exact same problem in all of them.

```
# MLProject file

name: test_norm_upload
docker_env:
  image: laerciop/nlptools_pipe_env:latest
  volumes: ["/home/laertio/projects/dev/nlp_tools/components/test_norm_upload/data_in:/data",
            "/home/laertio/projects/dev/nlp_tools/conf:/conf"]

entry_points:
  main:
    parameters:
      artifact_name:
        description: Name to identify your upload inside the NLP tank/W&B.
        type: string

      artifact_desc:
        description: Describe the data you're tryng to input. Please see the docs.
        type: string

      file_name:
        description: Name of the file to be placed in /data_in folder
        type: string

      database_type:
        description: Reference to the file source (read docs) to be tested
        type: string
      
      standard:
        description: Check if input follow standard or not.
        type: string
        default: in

      wandblogin:
        description: Wandb API key
        type: str

    command: python run.py  --artifact_name {artifact_name} --artifact_desc {artifact_desc} --file_name {file_name} --database_type {database_type} --standard {standard} --wandblogin {wandblogin}

# Dockerfile

FROM python:3.7

RUN apt update && \
    apt upgrade -y && \
    apt install -y curl \
      unzip \
      wget 


ADD requirements.txt /

RUN pip install -r requirements.txt

RUN python -m spacy download pt_core_news_sm

RUN  python -m spacy download pt_core_news_lg

```

### Stack trace

Output after "mlflow run ." in MLproject file directory.

```
2023/03/22 13:23:35 INFO mlflow.projects.docker: laerciop/nlptools_pipe_env:latest already exists
2023/03/22 13:23:35 INFO mlflow.projects.utils: === Created directory /tmp/tmp7c5g9ev2 for downloading remote URIs passed to arguments of type 'path' ===
2023/03/22 13:23:35 INFO mlflow.projects.backend.local: === Running command 'docker run --rm -v /home/laertio/projects/dev/nlp_tools/mlruns:/mlflow/tmp/mlruns -v /home/laertio/projects/dev/nlp_tools/mlruns/0/e251b38347b041e78e2e543d6e9f5638/artifacts:/home/laertio/projects/dev/nlp_tools/mlruns/0/e251b38347b041e78e2e543d6e9f5638/artifacts -v /home/laertio/projects/dev/nlp_tools/components/test_norm_upload/data_in:/data -v /home/laertio/projects/dev/nlp_tools/conf:/conf -e MLFLOW_RUN_ID=e251b38347b041e78e2e543d6e9f5638 -e MLFLOW_TRACKING_URI=file:///mlflow/tmp/mlruns -e MLFLOW_EXPERIMENT_ID=0 laerciop/nlptools_pipe_env:latest python run.py  --artifact_name test_musa --artifact_desc 'test for musa input' --file_name cervejamusa_ig.csv --database_type st_ig --standard in --wandblogin ea7d4ed782356f66850586612f115565b6a8a0c3' in run with ID 'e251b38347b041e78e2e543d6e9f5638' === 
python: can't open file 'run.py': [Errno 2] No such file or directory
2023/03/22 13:23:36 ERROR mlflow.cli: === Run (ID 'e251b38347b041e78e2e543d6e9f5638') failed ===
```

### Other info / logs

Here I provide, not the logs, but the files I have in the component folder (ls -l):

```
-rw-r--r-- 1 laertio adm 1265 Mar 22 13:23 MLproject
-rw-r--r-- 1 laertio adm   37 Mar 21 10:20 __init__.py
drwxr-xr-x 2 laertio adm 4096 Mar 20 17:49 data_in
drwxr-xr-x 2 laertio adm 4096 Mar 21 10:20 helpers
drwxr-xr-x 3 laertio adm 4096 Mar 20 17:41 mlruns
-rw-r--r-- 1 laertio adm 5177 Mar 21 12:30 run.py

```

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [X] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [X] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by laerciop**
- Simple changes in documentation (MLproject documentation)

## Related Issues/PRs

Close #8077 

## What changes are proposed in this pull request?

Change a documentation text to be more clear about a behaviour that changed between versions.

## How is this patch tested?

No test were needed since the PR refers to a documentation.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@laerciop The DCO check is not fixed yet. Can you fix it?

File: docs/source/projects.rst [hunk 1]
```diff
@@ -90,10 +90,12 @@ Docker container environment
...
-  When you run an MLflow project that specifies a Docker image, MLflow adds a new Docker layer
-  that copies the project's contents into the ``/mlflow/projects/code`` directory. This step produces
-  a new image. MLflow then runs the new image and invokes the project entrypoint in the resulting
-  container.
+  When you run an MLflow project that specifies a Docker image, MLflow runs your image as is with the parameters 
+  specified in your MLproject file. In this case you'll need to pre build your images with both environment 
+  and code to run it. In the oldest Mlflow versions, when specifying a docker environment mlflow, a new Docker layer
+  was added copying the project's contents into the ``/mlflow/projects/code`` directory producing a new image. 
+  Using ``mlflow run --build-image /path/to/directory`` makes MLflow then run the new image and invoke the
+  project entrypoint in the resulting container.
```
Comment (harupy) [inline docs/source/projects.rst:98]:
```suggestion
  When you run an MLflow project that specifies a Docker image, MLflow runs your image as is with the parameters 
  specified in your MLproject file. In this case you'll need to pre build your images with both environment 
  and code to run it. To use a new image that's based on your image and contains the project's contents into the ``/mlflow/projects/code`` directory, use the `--build-image` flag when running `mlflow run`.
```

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by laerciop** [comment]
Great, thanks for all the assistance @harupy  :)

**Pull request merged**